src/CoreWCF.Http/tests/Services/DuplexService.cs

```diff
@@ -0,0 +1,155 @@
+﻿using CoreWCF;
+using Helpers;
+using ServiceContract;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading;
+
+namespace Services
+{    
+    [ServiceBehavior(InstanceContextMode = InstanceContextMode.Single, ConcurrencyMode = ConcurrencyMode.Multiple,IncludeExceptionDetailInFaults =true)]
```

Comment:
@mconnew when ConcurrencyMode set to Reentrant, test show FaultException, Multiple and Single mode is OK, could you help to confirm is this a bug?
 Message: 
    System.ServiceModel.FaultException`1[[System.ServiceModel.ExceptionDetail, System.ServiceModel, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]] : Reentrant
  Stack Trace: 
    Server stack trace: 
    ServiceChannel.ThrowIfFaultUnderstood(Message reply, MessageFault fault, String action, MessageVersion version, FaultConverter faultConverter)
    ServiceChannel.HandleReply(ProxyOperationRuntime operation, ProxyRpc& rpc)
    ServiceChannel.Call(String action, Boolean oneway, ProxyOperationRuntime operation, Object[] ins, Object[] outs, TimeSpan timeout)
    ServiceChannelProxy.InvokeService(IMethodCallMessage methodCall, ProxyOperationRuntime operation)
    ServiceChannelProxy.Invoke(IMessage message)
    Exception rethrown at [0]: 
    RealProxy.HandleReturnMessage(IMessage reqMsg, IMessage retMsg)
    RealProxy.PrivateInvoke(MessageData& msgData, Int32 type)
    IDuplexService.UploadData(String data)
    AuthenticationTests.WindowsProxyAuthenticationSelfhostSecure() line 36